sympy/functions/elementary/piecewise.py

```diff
@@ -1018,61 +1023,24 @@ def _clip(A, B, k):
     interval (3, 4) was not covered by (1, 3) and is keyed to -1.
     """
     a, b = B
+    a, b = Min(a, b), b
```

Comment:
Put this after the c, d contraction since Min(Max(c, Min(a, b)), b) -> Min(Max(c, a), b)